public/docs/ts/latest/guide/style-guide.jade

```diff
@@ -0,0 +1,672 @@
+include ../_util-fns
+
+:marked
+  Welcome to the Angular 2 Guide of Style
+  
+  ## Purpose
+
+  If you are looking for an opinionated style guide for syntax, conventions, and structuring Angular applications, then step right in.
+
+  The purpose of this style guide is to provide guidance on building Angular applications by showing the conventions we use and, more importantly, why we choose them.
+.l-main-section
+a(id='toc')
+:marked
+  ## Table of Contents
+
+    1. [Single Responsibility](#single-responsibility)
+    1. [Modules](#modules)
+    1. [Components](#components)
+    1. [Services](#services)
+    1. [Data Services](#data-services)
+    1. [Naming](#naming)
+    1. [Application Structure LIFT Principle](#application-structure-lift-principle)
+    1. [Application Structure](#application-structure)
+    1. [Appendix](#appendix)
+:marked
+  ## Single Responsibility
+
+  ### Rule of 1 
+  <a id="001"></a>
+  #### Style 001
+
+  Define 1 component per file, recommended to be less than 400 lines of code.
+
+  **Why?** One component per file promotes easier unit testing and mocking.
+
+  **Why?** One component per file makes it far easier to read, maintain, and avoid collisions with teams in source control.
+
+  **Why?** One component per file avoids hidden bugs that often arise when combining components in a file where they may share variables, create unwanted closures, or unwanted coupling with dependencies.
+  
+  The following example defines the AppComponent, handles the bootstrapping, and shared functions all in the same file.
+
+  The key is to make the code more reusable, easier to read, and less mistake prone.
+
++makeExample('style-guide/ts/001/app/placeholder.ts', '001-1', 'app.component.ts (component)')
+:marked
+  The same components are now separated into their own files.
+
++makeTabs(
+  'style-guide/ts/001/app/main.ts,style-guide/ts/001/app/app.component.ts,style-guide/ts/001/app/hero.service.ts,style-guide/ts/001/app/hero.model.ts', 
+  '', 
+  'app/main.ts, app/app.component.ts, app/hero.service.ts, app/hero.model.ts')  
+:marked
+  As the app grows, this rule becomes even more important.
+
+<a href="#toc">Back to top</a>
+
+:marked
+  ### Small Functions
+  <a id="002"></a>
+  #### Style 002
+
+  - Define small functions, no more than 75 LOC (less is better).
+
+  **Why?** Small functions are easier to test, especially when they do one thing and serve one purpose.
+
+  **Why?** Small functions promote reuse.
+
+  **Why?** Small functions are easier to read.
+
+  **Why?** Small functions are easier to maintain.
+
+  **Why?** Small functions help avoid hidden bugs that come with large functions that share variables with external scope, create unwanted closures, or unwanted coupling with dependencies.
+
+<a href="#toc">Back to top</a>
+
+:marked
+  ## Modules
+
+  ### Avoid Naming Collisions
+  <a id="020"></a>
+  #### Style 020
+
+    - Use unique naming conventions.
```

Comment:
the conventions themselves should be unique? unique for each project or each file? that sounds a bit counter-productive to our goal. :)